src/pyclaw/solver.py

```diff
@@ -590,9 +590,10 @@ def evolve_to_time(self,solution,tend=None):
                     raise Exception('dt does not divide (tend-tstart) and dt is fixed!')
         if self.dt_variable == 1 and self.cfl_desired > self.cfl_max:
             raise Exception('Variable time-stepping and desired CFL > maximum CFL')
-        if tend <= tstart and not take_one_step:
-            self.logger.info("Already at or beyond end time: no evolution required.")
-            self.max_steps = 0
+        if not take_one_step:
+            if tend <= tstart and not take_one_step:
```

Comment:
That was a mistake, we need them separated but we don't want to check that twice.